File: app/src/main/java/com/hwonchul/movie/data/local/converter/LocalDateConverter.java
package com.hwonchul.movie.data.local.converter;

import androidx.annotation.Nullable;
import androidx.room.ProvidedTypeConverter;
import androidx.room.TypeConverter;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Optional;

@ProvidedTypeConverter
public class LocalDateConverter {
    @TypeConverter
    public LocalDate stringToLocalDate(@Nullable String date) {
        return Optional.ofNullable(date)
                .map(val -> LocalDate.parse(date, DateTimeFormatter.ofPattern("yyyy-MM-dd")))
                .orElse(null);
    }

    @TypeConverter
    public String localDateToString(@Nullable LocalDate date) {
        return Optional.ofNullable(date)
                .map(LocalDate::toString)
                .orElse(null);
    }
}

File: app/src/main/java/com/hwonchul/movie/data/local/converter/LocalDateConverter.kt
package com.hwonchul.movie.data.local.converter

import androidx.room.ProvidedTypeConverter
import androidx.room.TypeConverter
import java.time.LocalDate
import java.time.format.DateTimeFormatter

@ProvidedTypeConverter
class LocalDateConverter {
    @TypeConverter
    fun stringToLocalDate(date: String?): LocalDate? {
        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
        return date?.let { LocalDate.parse(date, formatter) }
    }

    @TypeConverter
    fun localDateToString(date: LocalDate?): String? {
        return date?.toString()
    }
}